File: js/axios/createWebhook.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;
async function createWebhook(webhookData) {
  try {
    return axios
      .post(apiUrl + "/webhooks", webhookData, {
        headers: {
          ContentType: "application/json",
          "x-api-key": accessKey,
        },
      })
      .then(function (response) {
        return response;
      })
      .catch(function (error) {
        console.log(error);
      });
  } catch (error) {
    // handle error
  }
}

const webhookData = {
  name: "TEST_HOOK",
  url: "https://webhook.site/a61857c1-ead8-4c2a-86a3-ebc910db2271",
  projectId: process.env.PROJECT_ID,
  topics: [
    "api-key.created",
    "api-key.deleted",
    "api-key.updated",
    "api-key.firstUsed",
    "access-key.created",
    "access-key.updated",
    "access-key.deleted",
    "project.updated",
  ], // list of webhook events with a min length of 1
  httpMethod: "POST", // http method of the request sent to the webhook
  customHeaders: {
    // custom headers added to the request sent to the webhook
    //a_random_number: "WEBHOOK-UID-" + Math.random(),
  },
  status: "live", // can be "live" or "paused"
};
(async () => {
  const createdWebhook = await createWebhook(webhookData);
  console.log(createdWebhook.data, createdWebhook.headers);
})();

File: python/webhooks/create_webhook.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/webhooks"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def create_webhook(create_webhook_data):
    try:
        response = requests.post(api_url, json=create_webhook_data, headers=headers)
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


create_webhook_data = {
    'name': 'Test webhook',
    'url': 'https://webhook.site/a61857c1-ead8-4c2a-86a3-ebc910db2271',
    'projectId': 'REPLACE_WITH_YOUR_PROJECT_ID',
    'topics': ['api-key.created', 'api-key.firstUsed'],
    'httpMethod': 'POST',
    'customHeaders': {
        # custom headers added to the request sent to the webhook
    },
    'status': 'live'
}
created_webhook = create_webhook(create_webhook_data)
print(created_webhook)
